src/Unreal.Core/BitReader.cs

```diff
@@ -104,27 +104,38 @@ public override ReadOnlySpan<byte> ReadBits(int bitCount)
             }
 
             var bitCountUsedInByte = Position & 7;
-            if (bitCountUsedInByte == 0 && bitCount % 8 == 0)
+            var byteCount = bitCount / 8;
+            var extraBits = bitCount % 8;
+            if (bitCountUsedInByte == 0 && extraBits == 0)
             {
-                return ReadBytes(bitCount >> 3);
+                var byteResult = Buffer.Span.Slice(CurrentByte, byteCount);
```

Comment:
The data/bits lines up perfectly, just grab the slice directly from the span and return it